tests/50federation/33room-get-missing-events.pl

```diff
@@ -517,21 +536,46 @@ sub sytest_user_and_room_fixture {
          $outbound_client->send_transaction(
             destination => $first_home_server,
             pdus => [ $sent_event ],
-         )->then( sub {
-            my ( $body ) = @_;
+         ),
+      )->then( sub {
+         log_if_fail "Sending marker event and waiting for /get_missing_events";
+
+         Future->needs_all(
+            $inbound_server->await_request_get_missing_events( $room_id )
+            ->then( sub {
+               my ( $req ) = @_;
+               my $body = $req->body_from_json;
+
+               log_if_fail "Second /get_missing_events body", $body;
 
-            log_if_fail "Send response", $body;
+               assert_json_list( my $earliest = $body->{earliest_events} );
+               @$earliest == 1 or
+                  die "Expected a single 'earliest_event' ID";
 
-            assert_json_keys( $body, 'pdus' );
-            # 'pdus' is a map from event id to error details.
-            my $pdus = $body->{pdus};
+               # It is expected that the earliest event is the m.room.member event,
+               # but it is possible that the caches have not yet been invalidated
+               # so also allow any of that event's previous events.
+               my @expected = @{$latest_event->{prev_events}};
+               push( @expected, $room->id_for_event( $latest_event ) );
+               assert_ok( any { $earliest->[0] eq $_ } @expected,
+                  "'earliest_events' did not match" );
 
-            # Sending the event fails since fetching the event results in
-            # invalid JSON, thus we expect an error for the sent PDU.
-            assert_json_keys( $pdus, $sent_event_id );
-            assert_json_keys( $pdus->{$sent_event_id}, qw( error ) );
+               assert_json_list( my $latest = $body->{latest_events} );
+               @$latest == 1 or
+                  die "Expected a single 'latest_events' ID";
+               assert_eq( $latest->[0], $marker_event_id,
+                  'latest_events[0]' );
 
-            Future->done;
-         }),
-      );
+               $req->respond_json( {
+                  events => [ $sent_event ],
+               } );
+
+               Future->done;
```

Comment:
if this is a cut and paste of the earlier code, let's factor it out to a separate `sub` ?